File: kotlin/src/main/java/com/kotlin/KTActivity.java
package com.kotlin;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

public class KTActivity extends AppCompatActivity {

    public  static String URI = CommonDefine.INSTANCE.getHOST()+"KTActivity";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.kotlin_main);
    }
}

File: kotlin/src/main/java/com/kotlin/KTActivity.kt
package com.kotlin

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import kotlinx.android.synthetic.main.kotlin_main.*

class KTActivity : AppCompatActivity() {

    companion object {
        var URI = CommonDefine.HOST + "KTActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.kotlin_main)

        btn_test.setOnClickListener(View.OnClickListener {
            startActivity(Intent(this, TestActivity::class.java))
            Thread(Runnable {
                try {
                    Thread.sleep(2000)
                    MyViewModel.init(application).post(AccountBean("lining", "1234567"))
                }catch (e: Exception){
                }
            }).start()
        })

        MyViewModel.init(application).observe(this, Observer<AccountBean> {
            tv_content.text = it?.id + it?.name
        })
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
        Log.i("lining", "-KTActivity--onBackPressed-----")
    }

}
